### PR TITLE
chore(docker): Fix docker-compose setup

### PR DESCRIPTION
**Pull request by lukkor**
Why? Docker improves and ease new contributors onboarding.

I updated node version at the same time pre-empting #313 (I needed to
update it in the docker for the setup to work).

As the #313 PR, it needs to be validated with a review app.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,6 +4,10 @@
 # or operating system, you probably want to add a global ignore instead:
 #   git config --global core.excludesfile '~/.gitignore_global'
 
+# Ignore yarn bin as prescribed by scalingo at
+# https://doc.scalingo.com/languages/ruby/rails/start#specifying-a-custom-nodejs-runtime
+/bin/yarn
```
Comment (lukkor) [inline .gitignore:9]: As recommended by the scalingo documentation, I removed yarn binary and ignore it for everyone not using docker can reinstall it.

**Comment by lukkor** [comment]
I changed the setup to only run databases in docker.
It is easier to use and less prompt to errors (files refreshed in volumes, ...).

**Comment by rchampourlier** [comment]
Tried to deploy on a review app.

Like #313 it may be needed to setup multiple buildpacks for it to work.

**Comment by lukkor** [comment]
@rchampourlier I updated the branch with only `docker-compose.yaml` file changes. It should not influence the deploy at all from what I can understand from the infra.
What do you think?

**Comment by kelu27** [comment]
I don't get why the web service is removed?

**Comment by rchampourlier** [comment]
- [ ] Just to be sure, confirm scalingo is ignoring the docker-compose

**Comment by lukkor** [comment]
@kelu27 Multiple reasons:

- The Dockerfile is not working as is (not hard to fix but it have to be done).
- In its current state, it is only "linking" the `app` directory from the host into the container whereas we are regularly modifying files in other directories too (e.g. `config/routes.rb`) and it is not optimized to rebuild the image every time we modify one of these files.
- IMO, Rails developers are used to run the application, console and tasks directly on the host so the Docker setup is usually used to run runtime dependencies like databases.

**Comment by lukkor** [comment]
@rchampourlier from what I can see from the docs, it is not using `docker-compose`.

File: README.md [hunk 2]
```diff
@@ -52,7 +52,8 @@ echo "LOCKBOX_MASTER_KEY=0000000000000000000000000000000000000000000000000000000
 
 ### Database / Cache
 
-1. Create a database called `covidliste_development` using your favorite postgresql GUI or CLI.
+1. Create a database called `covidliste_development` using your favorite
+   postgresql GUI or CLI.
```
Comment (EmCousin) [inline README.md:56]: We shoud avoid the manual step here anf follow along with the configuration setup in `config/database.yml`. There 's indeed a rake command dedicated to this step: `bin/rails db:create`

File: README.md [hunk 3]
```diff
@@ -52,7 +52,8 @@ echo "LOCKBOX_MASTER_KEY=0000000000000000000000000000000000000000000000000000000
...
 2. Then run the migrations : `bin/rails db:migrate RAILS_ENV=development`
```
Comment (EmCousin) [inline README.md:57]: We could just run `bin/setup` which handles all of this.

**Review by EmCousin — COMMENTED** [review]
Some suggestions, although they are not directly Docker-related

File: README.md (hunk 2 above)
Comment (rchampourlier) [reply README.md:56]: Agreed. I'll merge this one so we may go on and open a basic PR including your suggestions.

File: README.md (hunk 3 above)
Comment (rchampourlier) [reply README.md:57]: :+1:

**Review by rchampourlier — APPROVED** [review]

**Review by rchampourlier — APPROVED** [review]

**Comment by rchampourlier** [comment]
@kelu27 If you need to run the web app in the Docker, which I assume is not the case for most, you may probably use the docker-compose override feature (doc [here](https://docs.docker.com/compose/extends/)).

**Pull request merged**